### PR TITLE
(0014074) Implement new resource aviatrix_external_device_conn

### PR DESCRIPTION
**Pull request by zhiqiangjin**

### CODE REVIEW

File: aviatrix/provider.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ func Provider() terraform.ResourceProvider {
 			"aviatrix_azure_peer":                           resourceAviatrixAzurePeer(),
 			"aviatrix_azure_spoke_native_peering":           resourceAviatrixAzureSpokeNativePeering(),
 			"aviatrix_controller_config":                    resourceAviatrixControllerConfig(),
+			"aviatrix_external_device_conn":                 resourceAviatrixExternalDeviceConn(),
```
Comment (arvindsree) [inline aviatrix/provider.go:53]: Maybe rename to aviatrix_transit_gw_external_external_device_conn

File: aviatrix/resource_aviatrix_external_device_conn.go [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+				Description: "Remote CIDRs joined as a string with ','. Required for a static VPN connection.",
+			},
+			"pre_shared_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				ForceNew:    true,
+				Description: "Remote CIDRs joined as a string with ','. Required for a static VPN connection.",
+			},
+			"local_tunnel_ip": {
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_external_device_conn.go]: local_tunnel_cidr?

File: aviatrix/resource_aviatrix_external_device_conn.go [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Remote CIDRs joined as a string with ','. Required for a static VPN connection.",
+			},
+			"remote_tunnel_ip": {
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_external_device_conn.go]: remote_tunnel_cidr?

File: aviatrix/resource_aviatrix_external_device_conn.go [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+			},
+			"backup_pre_shared_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Sensitive:   true,
+				ForceNew:    true,
+				Description: "ID of the vpn connection.",
+			},
+			"backup_local_tunnel_ip": {
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_external_device_conn.go]: cidr

File: aviatrix/resource_aviatrix_external_device_conn.go [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				ForceNew:    true,
+				Description: "ID of the vpn connection.",
+			},
+			"backup_remote_tunnel_ip": {
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_external_device_conn.go]: cidr

File: aviatrix/resource_aviatrix_external_device_conn.go [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+		}
+	}
+
+	err = client.CreateExternalDeviceConn(externalDeviceConn)
+	if err != nil {
+		return fmt.Errorf("failed to create Aviatrix external device connection: %s", err)
+	}
+
+	d.SetId(externalDeviceConn.ConnName + "~" + externalDeviceConn.VpcID)
+	return resourceAviatrixExternalDeviceConnRead(d, meta)
```
Comment (arvindsree) [inline aviatrix/resource_aviatrix_external_device_conn.go:308]: again, require the defred read as  multiple operations are  performed above

Comment (zhiqiangjin) [reply aviatrix/resource_aviatrix_external_device_conn.go:308]: Here, only one operations is performed.

**Pull request merged**